lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2350,34 +2346,29 @@ void TFDeabstraction::evaluateAttributesAndDoPacking(
           if (unpackTensorAggregates(context, origInst->getLoc(), B, element,
                                      loweredTupleElts, loweredStructFields,
                                      unwrapCache, inputList)) {
-            diagnoseInvalid(argumentLoc,
-                            "argument of type '" + elementType->getString() +
-                            "' is not a TensorFlow value or an aggregate of "
-                            "TensorFlow values");
-            return;
+            noClustering = true;
+            break;
           }
         }
-        opBuilder.addListArgument(inputList);
+        if (noClustering)
```

Comment:
This prevents it from unpacking some arguments that it can unpack. For example, consider `#tfop("SomeOp", a, b)` where `a` cannot be unpacked at compile time and `b` can be unpacked at compile time. When it processes `a`, it'll set `noClustering = true`, which will cause it to add the non-unpacked version of `b` when it processes `b`. I suggest a new flag to keep track of whether it successfully unpacked this particular argument.

Reply:
This is a very interesting point. I thought more about this, and wasn't able to come up with such an example. 

I added the following comment block as an informal argument on why this couldn't happen. PTAL.
```
            // If some element of the array fails to unpack, we expect all
            // elements to fail to unpack (as they share the same array element
            // type), so we bail out of the loop over array elements, and call
            // `opBuilder.addArgument(argumentValue)` below.
```